src/test/base_test.hpp

```diff
@@ -112,15 +112,28 @@ class BaseTestWithParam
     if (predicate_condition == PredicateCondition::IsNull || predicate_condition == PredicateCondition::IsNotNull) {
       predicate = std::make_shared<IsNullExpression>(predicate_condition, column_expression);
     } else if (predicate_condition == PredicateCondition::Between) {
-      Assert(value2, "Need value2 for BetweenExpression");
-      predicate = std::make_shared<BetweenExpression>(column_expression, value_(value), value_(*value2));
+      return create_between_table_scan(in, column_id, value, value2);
     } else {
       predicate = std::make_shared<BinaryPredicateExpression>(predicate_condition, column_expression, value_(value));
     }
 
     return std::make_shared<TableScan>(in, predicate);
   }
 
+  // Utility to create between table scans
+  static std::shared_ptr<TableScan> create_between_table_scan(
```

Comment:
What is the advantage of making this static?
If it just shouldn't be used from outside the class, wouldn't a protected function work?
(We aren't sure how static function work in combination with subclasses)

Reply:
static just means, it does not depend on any object variables. Therefore, it is moved into the static part of the class. And we just followed the pattern of utility functions above and below, which are all tagged static ;)

Reply:
yes, thanks for the explanation